### DISCUSSION TITLE
Is Tarui zero-dependency in Production?

### DISCUSSION THREAD

**Discussion by axetroy** [tauri-apps/tauri#2924] [Q&A] [answered]
For many teams, before using it, we must figure out a few things:

1. System dependencies (Updater/Notifier/Dialog/Tray)
2. Runtime dependencies (eg. MS lib in Windows)
3. System webview required?

Since it is compared with Electron, this is a very important factor. If the system cannot be installed/used because the system lacks dependencies, and the user needs to install the dependencies by himself, then it will be catastrophic

**Comment by FabianLars** [comment, marked as answer]
Linux:
- .deb declares its dependencies so the package manager will install them.
- AppIamge is self-contained as always. That means it includes all needed dependencies in itself.
  - Currently there are 1 or 2 bugs preventing them from being 100% cross-distro: [[1]](https://github.com/tauri-apps/tauri/issues/2689) [[2]](https://github.com/tauri-apps/tauri/issues/2805)
- as a side note, building on linux generally requires to _build on the oldest system you want to support_, mainly because of glibc :))) But this can cause problems with the webkit2gtk version (regarding important bug fixes).

macOS:
- depends on the system webkit, which should be always present (== afaik there isn't a way to remove it from the system)
  - obviously the webkit version on the users system dictates which html/css/js features are available.

windows:
- needs webview2 installed (at runtime)
  - preinstalled on windows 11
  - win7/8/10: the msi takes care of the installation, but the user has to accept the UAC prompt for the webview2 installer.
- needs a c++ redistributable (at runtime), also included in the msi

**Comment by amrbashir** [comment]
A final Tauri app will depend on a few things to exist on the  user system for it to work properly.

- On Windows, you need `vcredist14` and `Webview2`, most Windows systems will have vcredist installed as it is a common dependency for other apps too and Webview2 is pre-shipped in Win11 and hopefully Win10 will too but our bundler will still generate an `.msi` installer that will handle both of these dependencies (it will download the `Webview2` installer and install it if needed).

- On linux, you need `webkit2gtk` and `libappindicator` and `gtk` which are probably installed on most Linux systems but our bundler will also take care of it if they are not, it will generate a `.deb` package that specifies its dependencies so the package manager will install them if they don't exist. Also our bundler can make an `Appimage` that has all the dependencies included (it is a bit buggy right now but it will get better). except `gtk` I think because it is usually already installed on Linux.

- On macOS, Afaik, nothing is needed, it will just work out of the box. 

Edit: Apologies, didn't see that @FabianLars already responded with a beautiful detailed answer.
